### ISSUE TITLE
In the case of `spring-data-redis 1.4.2`, which associated with the Jedis version of the case of `redis.clients 2.5.2`, in the monitoring tomcat` catalina.out`, the error is:` ClassInstanceMethodsInterceptor: class [class + redis.clients.jedis.Jedis ] Before method [disconnect] intercept failure.`

### ISSUE THREAD

**Issue by WallC0902** [wu-sheng/sky-walking#150]
In the case of `spring-data-redis 1.4.2`, which associated with the Jedis version of the case of `redis.clients 2.5.2`, in the monitoring tomcat` catalina.out`, the error is:` ClassInstanceMethodsInterceptor: class [class + redis.clients.jedis.Jedis ] Before method [disconnect] intercept failure.`
the detail error:
```

ERROR 2017-04-18 10:36:43 ClassInstanceMethodsInterceptor :  class[class+redis.clients.jedis.Jedis] before method[disconnect] intercept failure 
java.lang.IllegalAccessError: com/a/eye/skywalking/plugin/jedis/v2/JedisMethodInterceptor$1
	at com.a.eye.skywalking.plugin.jedis.v2.JedisMethodInterceptor.beforeMethod(JedisMethodInterceptor.java:47)
	at com.a.eye.skywalking.api.plugin.interceptor.enhance.ClassInstanceMethodsInterceptor.intercept(ClassInstanceMethodsInterceptor.java:56)
	at redis.clients.jedis.Jedis.disconnect(Jedis.java)
	at redis.clients.jedis.JedisFactory.destroyObject(JedisFactory.java:53)
	at org.apache.commons.pool2.impl.GenericObjectPool.destroy(GenericObjectPool.java:865)
	at org.apache.commons.pool2.impl.GenericObjectPool.evict(GenericObjectPool.java:775)
	at org.apache.commons.pool2.impl.BaseGenericObjectPool$Evictor.run(BaseGenericObjectPool.java:1045)
	at java.util.TimerThread.mainLoop(Timer.java:555)
	at java.util.TimerThread.run(Timer.java:505)

ERROR 2017-04-18 10:36:43 ClassInstanceMethodsInterceptor :  class[class+redis.clients.jedis.Jedis] after method[disconnect] intercept failure 
java.lang.IllegalAccessError: com/a/eye/skywalking/plugin/jedis/v2/JedisMethodInterceptor$2
	at com.a.eye.skywalking.plugin.jedis.v2.JedisMethodInterceptor.afterMethod(JedisMethodInterceptor.java:86)
	at com.a.eye.skywalking.api.plugin.interceptor.enhance.ClassInstanceMethodsInterceptor.intercept(ClassInstanceMethodsInterceptor.java:77)
	at redis.clients.jedis.Jedis.disconnect(Jedis.java)
	at redis.clients.jedis.JedisFactory.destroyObject(JedisFactory.java:53)
	at org.apache.commons.pool2.impl.GenericObjectPool.destroy(GenericObjectPool.java:865)
	at org.apache.commons.pool2.impl.GenericObjectPool.evict(GenericObjectPool.java:775)
	at org.apache.commons.pool2.impl.BaseGenericObjectPool$Evictor.run(BaseGenericObjectPool.java:1045)
	at java.util.TimerThread.mainLoop(Timer.java:555)
	at java.util.TimerThread.run(Timer.java:505)
```
Environment: centos 7, tomcat 8

**Comment by ascrutae** [comment]
I cannot reproduce this bug. Here is my example and call graph.

Call graph:
<img width="1422" alt="wx20170418-152602 2x" src="https://cloud.githubusercontent.com/assets/12436447/25119160/a0f83a42-244b-11e7-9aea-87c50d11094b.png">

example code:
[spring-redis-test.zip](https://github.com/wu-sheng/sky-walking/files/927677/spring-redis-test.zip)

**Comment by wu-sheng** [comment]
@WallC0902  Can you give us your demo for reproducing your issue? Hard to tell the reason if we can not reproduce.

I guess, maybe relate with your other libraries, JDV edition, or something useless. Maybe you can check @ascrutae 's codes, and give us more clues.

**Comment by WallC0902** [comment]
@wu-sheng In my environment, redis set the password (`mytijian`), I tested @ascrutae 's code, in the `connectionFactory` not set` p: password` field, the error：
```
19:42:52.044 [main] DEBUG o.s.d.r.core.RedisConnectionUtils - Opening RedisConnection
Exception in thread "main" org.springframework.data.redis.RedisConnectionFailureException: Cannot get Jedis connection; nested exception is redis.clients.jedis.exceptions.JedisConnectionException: 
	Could not get a resource from the pool
	at org.springframework.data.redis.connection.jedis.JedisConnectionFactory.fetchJedisConnector(JedisConnectionFactory.java:140)
	at org.springframework.data.redis.connection.jedis.JedisConnectionFactory.getConnection(JedisConnectionFactory.java:229)
	at org.springframework.data.redis.connection.jedis.JedisConnectionFactory.getConnection(JedisConnectionFactory.java:57)
	at org.springframework.data.redis.core.RedisConnectionUtils.doGetConnection(RedisConnectionUtils.java:128)
	at org.springframework.data.redis.core.RedisConnectionUtils.getConnection(RedisConnectionUtils.java:91)
	at org.springframework.data.redis.core.RedisConnectionUtils.getConnection(RedisConnectionUtils.java:78)
	at org.springframework.data.redis.core.RedisTemplate.execute(RedisTemplate.java:177)
	at org.springframework.data.redis.core.RedisTemplate.execute(RedisTemplate.java:152)
	at org.springframework.data.redis.core.RedisTemplate.execute(RedisTemplate.java:140)
	at com.a.eye.skywalking.spring.redis.TestDao.saveBean(TestDao.java:17)
	at com.a.eye.skywalking.spring.redis.Main.main(Main.java:14)
Caused by: redis.clients.jedis.exceptions.JedisConnectionException: Could not get a resource from the pool
	at redis.clients.util.Pool.getResource(Pool.java:42)
	at redis.clients.jedis.JedisPool.getResource(JedisPool.java:84)
	at redis.clients.jedis.JedisPool.getResource(JedisPool.java:10)
	at org.springframework.data.redis.connection.jedis.JedisConnectionFactory.fetchJedisConnector(JedisConnectionFactory.java:133)
	... 10 more
Caused by: java.util.NoSuchElementException: Unable to validate object
	at org.apache.commons.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:497)
	at org.apache.commons.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:360)
	at redis.clients.util.Pool.getResource(Pool.java:40)
	... 13 more
```
When I set it `p:password` up, 
```
<bean id="connectionFactory" class="org.springframework.data.redis.connection.jedis.JedisConnectionFactory"
          p:host-name="${redis.host}" p:port="${redis.port}" p:password="${redis.password}" p:pool-config-ref="poolConfig"/>
```
everything is fine.
```
19:44:35.988 [main] DEBUG o.s.d.r.core.RedisConnectionUtils - Opening RedisConnection
19:44:35.988 [main] DEBUG o.s.d.r.core.RedisConnectionUtils - Closing Redis Connection
test

Process finished with exit code 0

```

**Comment by WallC0902** [comment]
OK，I also re-check my code. Find other reasons

**Comment by wu-sheng** [comment]
I will leave this open until you solve your problem.😀

If you need any help, let us know.

**Comment by wu-sheng** [comment]
@WallC0902 Do you have some update?

**Comment by WallC0902** [comment]
Yes, the problem is always there. I'm sorting out the demo that can reproduce the problem.

**Comment by wu-sheng** [comment]
@WallC0902 Thanks. Hope we can solve this, after your demo provided.

**Comment by WallC0902** [comment]
@wu-sheng @ascrutae 
This is my demo, you can reproduce that problem。
[ops-manager-site.tar.gz](https://github.com/wu-sheng/sky-walking/files/960398/ops-manager-site.tar.gz)
use tomcat 8，and test path is: ip:port/action/test/testRedis?val=testRedis.

**Comment by wu-sheng** [comment]
@WallC0902 Thanks, cc @ascrutae you can try to reproduce this after we release 3.0.1

**Comment by ascrutae** [comment]
Got it.

**Comment by ascrutae** [comment]
The issue has been fix in 3.0.1.

**Comment by WallC0902** [comment]
OK , tks
